### PR TITLE
Add install from source instructions

### PR DESCRIPTION
**Pull request by davetcoleman**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -40,6 +40,11 @@ On other systems you can install bloom via pypi::
 
 Note: pip will not notify you of updates, but bloom will notify you when you are using a version of bloom that is not the latest released.
 
+From source::
+
+    $ python setup.py build
+    $ sudo python setup.py install
```
Comment (mikepurvis) [inline docs/index.rst]: I prefer `pip install git+https://github.com/ros-infrastructure/bloom.git`, but whatever floats your boat. :)

Comment (davetcoleman) [inline docs/index.rst]: Will that checkout the repo in the current directory such that one can make changes also?

Comment (mikepurvis) [inline docs/index.rst]: Nope, but in that case, perhaps it's better to recommend `sudo python setup.py develop`

Comment (davetcoleman) [inline docs/index.rst]: I presume that allows you to develop in-place via symlinking? I'm not a python dev...

Comment (wjwwood) [inline docs/index.rst]: If you do that then it should be labeled as "develop on bloom" not "install from source". I would recommend people use `develop` mode if they just want to install it from source.

Comment (wjwwood) [inline docs/index.rst]: Also, https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs.

Comment (davetcoleman) [inline docs/index.rst]: updated

**Comment by wjwwood** [comment]
Thanks @davetcoleman. @mikepurvis if you prefer the pip install syntax (with or without `-e`) I'd merge a pull request that used that instead. I don't have time right now myself.

**Pull request merged**